### PR TITLE
docs: add kusion module in the concept

### PR DESCRIPTION
**Pull request by SparkYuan**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [ ] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/kusion/3-concepts/3-kusion-module.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+sidebar_position: 4
+---
```
Comment (healthjyk) [inline docs/kusion/3-concepts/3-kusion-module.md:3]: Delete sidebar_position

File: docs/kusion/3-concepts/4-workspace.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ id: workspace
 
 ## Definition
 
-A workspace is a logical concept that represents a target environment for deploying a stack. It contains a set of configurations, Kubeconfig, and provider authentication information that can be reused by multiple stacks. We recommend organizing workspaces by SDLC (Software Development Life Cycle) phases or by cloud vendors. For example, workspaces could be named `dev`, `staging`, and `prod`, or according to cloud vendors such as `AWS`, `Azure`, and `Aliyun`.
+A workspace is a logical concept that represents a target environment for deploying a stack. It contains platform configurations, including a set of configurations, Kubeconfig, and provider authentication information, all of which can be reused by multiple stacks. We recommend organizing workspaces by SDLC (Software Development Life Cycle) phases or by cloud vendors. For example, workspaces could be named `dev`, `staging`, and `prod`, or according to cloud vendors such as `AWS`, `Azure`, and `Aliyun`.
```
Comment (healthjyk) [inline docs/kusion/3-concepts/4-workspace.md:9]: Not 'Aliyun', but 'Alicloud'

File: docs/kusion/3-concepts/3-kusion-module.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+- App developer-oriented schema: It is a [KCL schema](https://kcl-lang.io/docs/user_docs/guides/schema-definition/). Fields in this schema are recommended to be understandable to application developers and workspace-agnostic. For example, a database Kusion module schema only contains fields like database engine type and database version.
+- Kusion module generator: It is a piece of logic that generates the Intent with an instantiated schema mentioned above, along with platform configurations ([workspace](4-workspace.md)). As a building block, Kusion module hides the complexity of infrastructures. A database Kusion module not only represents a cloud RDS, but it also contains logic to configure other resources such as security groups and IAM policies. Additionally, it seamlessly injects the database host address, username, and password into the workload's environment variables. The generator logic can be very complex in some situations so we recommend implementing it in a GPL like [go](https://go.dev/).
```
Comment (healthjyk) [inline docs/kusion/3-concepts/3-kusion-module.md:9]: ref use [workspace]

Comment (SparkYuan) [reply docs/kusion/3-concepts/3-kusion-module.md:9]: I have tried to use workspace, but it didn't work

**Review by healthjyk — APPROVED** [review]

**Pull request merged**